commit 3c9cba8efaa82fd47a4ba05d894ecc60e0d01cd6
Author: Jonathon Fitch <lightman1@virginmedia.com>
Date:   Sat Dec 31 14:20:05 2016 +0000

    Revert "Revert "Temporary apply LIBC version to __pthread_gettid""
    
    This reverts commit bba395492a0bb6ee72d0ad8e4d468e852392220e.

diff --git a/libc/bionic/ndk_cruft.cpp b/libc/bionic/ndk_cruft.cpp
index 3ac88f8..602adb4 100644
--- a/libc/bionic/ndk_cruft.cpp
+++ b/libc/bionic/ndk_cruft.cpp
@@ -340,10 +340,17 @@ size_t dlmalloc_usable_size(void* ptr) {
 }
 
 // In L we added a public pthread_gettid_np, but some apps were using the private API.
-pid_t __pthread_gettid(pthread_t t) {
+pid_t __pthread_gettid_libc(pthread_t t) {
   return pthread_gettid_np(t);
 }
 
+pid_t __pthread_gettid_libc_private(pthread_t t) {
+  return pthread_gettid_np(t);
+}
+
+__asm__(".symver __pthread_gettid_libc,__pthread_gettid@LIBC");
+__asm__(".symver __pthread_gettid_libc_private,__pthread_gettid@@LIBC_PRIVATE");
+
 // Older versions of apportable used dlmalloc directly instead of malloc,
 // so export this compatibility shim that simply calls malloc.
 void* dlmalloc(size_t size) {
diff --git a/libc/libc.arm.map b/libc/libc.arm.map
index 38f8437..fc7d37a 100644
--- a/libc/libc.arm.map
+++ b/libc/libc.arm.map
@@ -122,6 +122,7 @@ LIBC {
     __pselect6; # arm x86 mips
     __pthread_cleanup_pop;
     __pthread_cleanup_push;
+    __pthread_gettid; # arm x86 mips nobrillo
     __ptrace; # arm x86 mips
     __putlong;
     __putshort;
diff --git a/libc/libc.map.txt b/libc/libc.map.txt
index 0e41f6c..3e87dad 100644
--- a/libc/libc.map.txt
+++ b/libc/libc.map.txt
@@ -122,6 +122,7 @@ LIBC {
     __pselect6; # arm x86 mips
     __pthread_cleanup_pop;
     __pthread_cleanup_push;
+    __pthread_gettid; # arm x86 mips nobrillo
     __ptrace; # arm x86 mips
     __putlong;
     __putshort;
diff --git a/libc/libc.mips.map b/libc/libc.mips.map
index 46c835b..3a2557f 100644
--- a/libc/libc.mips.map
+++ b/libc/libc.mips.map
@@ -119,6 +119,7 @@ LIBC {
     __pselect6; # arm x86 mips
     __pthread_cleanup_pop;
     __pthread_cleanup_push;
+    __pthread_gettid; # arm x86 mips nobrillo
     __ptrace; # arm x86 mips
     __putlong;
     __putshort;
diff --git a/libc/libc.x86.map b/libc/libc.x86.map
index 9417d56..a2e8586 100644
--- a/libc/libc.x86.map
+++ b/libc/libc.x86.map
@@ -119,6 +119,7 @@ LIBC {
     __pselect6; # arm x86 mips
     __pthread_cleanup_pop;
     __pthread_cleanup_push;
+    __pthread_gettid; # arm x86 mips nobrillo
     __ptrace; # arm x86 mips
     __putlong;
     __putshort;
